src/main/java/net/glowstone/block/blocktype/BlockType.java

```diff
@@ -369,7 +352,19 @@ public final void rightClickBlock(GlowPlayer player, GlowBlock against, BlockFac
         }
 
         // call canBuild event
-        boolean canBuild = canPlaceAt(target, face);
+        boolean canBuild = true;
```

Comment:
Revert this, and then you can use a simple `if (!player.isSneaking())` with no `else`.